### PR TITLE
added user controller

### PR DESCRIPTION
**Pull request by maybee16**
https://trello.com/c/AkeaaU56/20-educationservice-add-user-controller

### CODE REVIEW

File: src/EducationService.Business/Commands/User/FindUsersCommand.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    if (!validationResult.IsValid)
+    {
+      return _responseCreator.CreateFailureFindResponse<EducationInfo>(HttpStatusCode.BadRequest,
+        validationResult.Errors.Select(vf => vf.ErrorMessage).ToList());
+    }
+
+    List<DbUserEducation> dbUserEducations = await _userEducationRepository.GetAsync(userId);
+
+    return new FindResultResponse<EducationInfo>(
+      body: dbUserEducations.Select(ue => _educationInfoMapper.Map(ue)).ToList());
```
Comment (VLiamin) [inline src/EducationService.Business/Commands/User/FindUsersCommand.cs]: May be remove ue?

File: src/EducationService.Models.Dto/Requests/User/FindUsersFilter.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using LT.DigitalOffice.Kernel.Requests;
+using Microsoft.AspNetCore.Mvc;
+using System;
+
+namespace LT.DigitalOffice.EducationService.Models.Dto.Requests.User;
+
+public record FindUsersFilter : BaseFindFilter
+{
+  [FromQuery(Name = "educationFormId")]
```
Comment (VLiamin) [inline src/EducationService.Models.Dto/Requests/User/FindUsersFilter.cs]: We usually write 'education**f**orm**i**d'

File: src/EducationService.Business/Commands/User/FindUsersCommand.cs (hunk 1 above)
Comment (maybee16) [reply src/EducationService.Business/Commands/User/FindUsersCommand.cs]: Why?

File: src/EducationService.Business/Commands/User/FindUsersCommand.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  {
+    ValidationResult validationResult = _baseFindFilterValidator.Validate(filter);
+
+    if (!validationResult.IsValid)
+    {
+      return _responseCreator.CreateFailureFindResponse<EducationInfo>(HttpStatusCode.BadRequest,
+        validationResult.Errors.Select(vf => vf.ErrorMessage).ToList());
+    }
+
+    List<DbUserEducation> dbUserEducations = await _userEducationRepository.GetAsync(userId);
```
Comment (MariyaSin) [inline src/EducationService.Business/Commands/User/FindUsersCommand.cs]: you should get userEducations by filter conditions

File: src/EducationService.Business/Commands/User/Interfaces/IFindUsersCommand.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+using LT.DigitalOffice.Kernel.Responses;
+using System;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.EducationService.Business.Commands.User.Interfaces;
+
+[AutoInject]
+public interface IFindUsersCommand
+{
+  Task<FindResultResponse<EducationInfo>> ExecuteAsync(Guid userId, FindUsersFilter filter);
```
Comment (MariyaSin) [inline src/EducationService.Business/Commands/User/Interfaces/IFindUsersCommand.cs]: you can put userId in filter

File: src/EducationService.Mappers/Models/EducationInfoMapper.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  {
+    return dbUserEducation is null
+      ? null
+      : new EducationInfo
+      {
+        Id = dbUserEducation.Id,
+        UserId = dbUserEducation.UserId,
+        UniversityName = dbUserEducation.UniversityName,
+        QualificationName = dbUserEducation.QualificationName,
+        EducationFormId = dbUserEducation.EducationFormId,
```
Comment (MariyaSin) [inline src/EducationService.Mappers/Models/EducationInfoMapper.cs]: map and put in response Form id and FormName.
Create EducationFormInfo model (Guid Id, string Name)

File: src/EducationService.Mappers/Models/EducationInfoMapper.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        EducationTypeId = dbUserEducation.EducationTypeId,
```
Comment (MariyaSin) [inline src/EducationService.Mappers/Models/EducationInfoMapper.cs]: same

File: src/EducationService.Models.Dto/Models/EducationInfo.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace LT.DigitalOffice.EducationService.Models.Dto.Models;
+
+public record EducationInfo
+{
+  public Guid Id { get; set; }
+  public Guid UserId { get; set; }
+  public string UniversityName { get; set; }
+  public string QualificationName { get; set; }
+  public Guid EducationFormId { get; set; }
```
Comment (MariyaSin) [inline src/EducationService.Models.Dto/Models/EducationInfo.cs]: public EducationFormInfo EducationForm

File: src/EducationService.Models.Dto/Models/EducationInfo.cs [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  public Guid EducationTypeId { get; set; }
```
Comment (MariyaSin) [inline src/EducationService.Models.Dto/Models/EducationInfo.cs]: public EducationTypeInfo EducationType

File: src/EducationService.Models.Dto/Requests/User/FindUsersFilter.cs [hunk 9]
```diff
@@ -0,0 +1,17 @@
+﻿using LT.DigitalOffice.Kernel.Requests;
+using Microsoft.AspNetCore.Mvc;
+using System;
+
+namespace LT.DigitalOffice.EducationService.Models.Dto.Requests.User;
+
+public record FindUsersFilter : BaseFindFilter
+{
```
Comment (MariyaSin) [inline src/EducationService.Models.Dto/Requests/User/FindUsersFilter.cs:9]: put UserId from query here

File: src/EducationService.Models.Dto/Requests/User/FindUsersFilter.cs (hunk 2 above)
Comment (maybee16) [reply src/EducationService.Models.Dto/Requests/User/FindUsersFilter.cs]: Done

File: src/EducationService.Business/Commands/User/FindUsersCommand.cs (hunk 3 above)
Comment (maybee16) [reply src/EducationService.Business/Commands/User/FindUsersCommand.cs]: Done

File: src/EducationService.Business/Commands/User/Interfaces/IFindUsersCommand.cs (hunk 4 above)
Comment (maybee16) [reply src/EducationService.Business/Commands/User/Interfaces/IFindUsersCommand.cs]: Done

File: src/EducationService.Mappers/Models/EducationInfoMapper.cs (hunk 5 above)
Comment (maybee16) [reply src/EducationService.Mappers/Models/EducationInfoMapper.cs]: Done

File: src/EducationService.Mappers/Models/EducationInfoMapper.cs (hunk 6 above)
Comment (maybee16) [reply src/EducationService.Mappers/Models/EducationInfoMapper.cs]: Done

File: src/EducationService.Models.Dto/Models/EducationInfo.cs (hunk 7 above)
Comment (maybee16) [reply src/EducationService.Models.Dto/Models/EducationInfo.cs]: Done

File: src/EducationService.Models.Dto/Models/EducationInfo.cs (hunk 8 above)
Comment (maybee16) [reply src/EducationService.Models.Dto/Models/EducationInfo.cs]: Done

File: src/EducationService.Models.Dto/Requests/User/FindUsersFilter.cs (hunk 9 above)
Comment (maybee16) [reply src/EducationService.Models.Dto/Requests/User/FindUsersFilter.cs:9]: Done

File: src/EducationService.Business/Commands/User/FindUsersCommand.cs (hunk 1 above)
Comment (maybee16) [reply src/EducationService.Business/Commands/User/FindUsersCommand.cs]: Done

File: src/EducationService.Models.Dto/Requests/User/FindUsersFilter.cs [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+  public Guid UserId { get; set; }
+
+  [FromQuery(Name = "educationformid")]
+  public Guid? EducationFormId { get; set; }
+
+  [FromQuery(Name = "educationtypeid")]
+  public Guid? EducationTypeId { get; set; }
+
+  [FromQuery(Name = "completeness")]
+  public int? Completeness { get; set; }
```
Comment (MariyaSin) [inline src/EducationService.Models.Dto/Requests/User/FindUsersFilter.cs]: use enum

Comment (maybee16) [reply src/EducationService.Models.Dto/Requests/User/FindUsersFilter.cs]: Done

**Review by MariyaSin — APPROVED** [review]

File: src/EducationService.Data/UserEducationRepository.cs [hunk 11]
```diff
@@ -46,5 +47,29 @@ public async Task DisactivateEducationsAsync(Guid userId, Guid modifiedBy)
...
+      {
+        query = query.Where(e => e.EducationType.Id == filter.EducationTypeId.Value);
+      }
+
+      if (filter.Completeness.HasValue)
+      {
+        query = query.Where(e => e.Completeness == (int)filter.Completeness.Value);
+      }
+
+      return await query.Skip(filter.SkipCount).Take(filter.TakeCount).ToListAsync();
```
Comment (AnnPostnikova) [inline src/EducationService.Data/UserEducationRepository.cs]: remove await and async, return Task
you`re already using await in command on 48 line

Comment (maybee16) [reply src/EducationService.Data/UserEducationRepository.cs]: Shouldn't the method working with the database be asynchronous?

Comment (maybee16) [reply src/EducationService.Data/UserEducationRepository.cs]: Done

**Review by AnnPostnikova — APPROVED** [review]

**Pull request merged**